linera-core/src/client.rs

```diff
@@ -598,16 +600,33 @@ where
             },
         )
         .await;
+        let (value, round) = match action {
+            CommunicateAction::SubmitBlockForConfirmation(proposal) => {
+                let block = proposal.content.block;
+                let (executed_block, _) = self.node_client.stage_block_execution(block).await?;
+                (
+                    HashedValue::from(CertificateValue::ConfirmedBlock { executed_block }),
+                    proposal.content.round,
+                )
+            }
+            CommunicateAction::SubmitBlockForValidation(proposal) => {
+                let BlockAndRound { block, round } = proposal.content;
+                let (executed_block, _) = self.node_client.stage_block_execution(block).await?;
+                (HashedValue::new_validated(executed_block), round)
+            }
+            CommunicateAction::FinalizeBlock(validity_certificate) => {
+                let round = validity_certificate.round;
+                (validity_certificate.value.into_confirmed(), round)
+            }
+            CommunicateAction::AdvanceToNextBlockHeight(_) => return Ok(None),
+        };
         let votes = match result {
-            Ok((_hash, votes)) => votes,
-            Err(CommunicationError::Trusted(NodeError::InactiveChain(id)))
```

Comment:
Are we removing this logic intentionally? This would deserve a separate commit and a word in the summary.

Reply:
This case is no longer reachable: We now return in line 621 if we're in the `AdvanceToNextBlockHeight` case, so the `matches!` below cannot be `true`.

Reply:
It seems that line 621 should analyze `result` to process the error cases as before instead of always succeeding.

Reply:
Right, good catch! Done.